File: leetcode/rust/maximum-nesting-depth-of-the-parentheses.rs
impl Solution {
    pub fn max_depth(s: String) -> i32 {
        let mut result = 0;
        let mut count = 0;

        for c in s.chars() {
            match c {
                '(' => count += 1,
                ')' => count -= 1,
                _ => (),
            };

            result = result.max(count);
        }

        result
    }
}

File: leetcode/python/maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        result = 0
        cur = 0

        for c in s:
            if c == '(':
                cur += 1
                result = max(result, cur)
            elif c == ')':
                cur -= 1
        
        return result
        